OpenRA.Game/Scripting/ScriptMemberWrapper.cs

```diff
@@ -141,5 +142,20 @@ public static IEnumerable<MemberInfo> WrappableMembers(Type t)
 				return false;
 			});
 		}
+
+		public static string[] RequiredTraitNames(Type t)
+		{
+			// Returns the inner types of all the Requires<T> interfaces on this type
+			var outer = t.GetInterfaces()
+				.Where(i => i.IsGenericType && i.GetGenericTypeDefinition() == typeof(Requires<>));
+
+			// Get the inner types
+			var inner = outer.SelectMany(i => i.GetGenericArguments()).ToArray();
```

Comment:
This is only iterated once, so it doesn't look like we need/want that `.ToArray` here.

Reply:
These should just be joined, maybe even with the `return` statement.